src/main/java/com/arpnetworking/tsdcore/sinks/WavefrontSink.java

```diff
@@ -0,0 +1,284 @@
+            reportPointBuilder.setValue(aggregatedData.getValue().getValue());
+            reportPointBuilder.setTimestamp(periodicData.getStart().getMillis());
+            reportPointBuilder.setTable("dummy");
+            _pointHandler.reportPoint(reportPointBuilder.build(), "");
+        }
+    }
+
+    @Override
+    public void close() {
+    }
+
+    private String buildMetricName(final FQDSN fqdsn, final Period period) {
+        return fqdsn.getService()
+                + "."
+                + period.toString(ISOPeriodFormat.standard())
+                + "."
+                + fqdsn.getMetric()
+                + "."
+                + fqdsn.getStatistic().getName();
+    }
+
+    private String[] buildAgentArgs() {
+        final List<String> argList = new ArrayList<>();
+
+        // Token parameter
+        argList.add("-t");
+        argList.add(_token);
+
+        // Host parameter
+        argList.add("-h");
+        argList.add(_uri.toString());
+
+        // Retry threads if set
+        if (_retryThreads != null) {
+            argList.add("--retryThreads");
+            argList.add(_retryThreads.toString());
+        }
+
+        // Flush threads if set
+        if (_flushThreads != null) {
+            argList.add("--flushThreads");
+            argList.add(_flushThreads.toString());
+        }
+
+        // Purge buffer
+        argList.add("--purgeBuffer");
+        argList.add(_purgeBuffer ? "true" : "false");
+
+        // Push Flush Interval if set
+        if (_pushFlushInterval != null) {
+            argList.add("--pushFlushInterval");
+            argList.add(_pushFlushInterval.toString());
+        }
+
+        // Push Flush Max Points if set
+        if (_pushFlushMaxPoints != null) {
+            argList.add("--pushFlushMaxPoints");
+            argList.add(_pushFlushMaxPoints.toString());
+        }
+
+        // Id File
+        argList.add("--idFile");
+        argList.add(_idFile);
+
+        return argList.toArray(new String[argList.size()]);
+    }
+
+    private WavefrontSink(final Builder builder) {
+        super(builder);
+        _uri = builder._uri;
+        _token = builder._token;
+        _retryThreads = builder._retryThreads;
+        _flushThreads = builder._flushThreads;
+        _purgeBuffer = builder._purgeBuffer;
+        _pushFlushInterval = builder._pushFlushInterval;
+        _pushFlushMaxPoints = builder._pushFlushMaxPoints;
+        _idFile = builder._idFile;
+        final WavefrontAgent agent = new WavefrontAgent();
+        try {
+            agent.start(buildAgentArgs());
+        } catch (final IOException e) {
+            LOGGER.warn()
+                    .setMessage("Error configuring Wavefront Agent")
+                    .setThrowable(e);
+
+        }
+
+        _pointHandler = agent.createPointHandler();
+    }
+
+    private final URI _uri;
+    private final String _token;
+    private final Integer _retryThreads;
```

Comment:
Really?  That's not IntelliJ's interpretation of how Optional should be used.  They added an inspection specifically flagging this as an incorrect use of Optional, they state that it was intended for library method return types and not fields as it's not Serializable.